lib/ClangImporter/ClangImporter.cpp

```diff
@@ -7741,14 +7817,14 @@ CxxRecordSemantics::evaluate(Evaluator &evaluator,
   if (!hasDestroyTypeOperations(cxxDecl) ||
       (!hasCopyTypeOperations(cxxDecl) && !hasMoveTypeOperations(cxxDecl))) {
     if (hasUnsafeAPIAttr(cxxDecl))
-      desc.ctx.Diags.diagnose({}, diag::api_pattern_attr_ignored,
-                              "import_unsafe", decl->getNameAsString());
+      ctx.Diags.diagnose({}, diag::api_pattern_attr_ignored, "import_unsafe",
+                         decl->getNameAsString());
     if (hasOwnedValueAttr(cxxDecl))
-      desc.ctx.Diags.diagnose({}, diag::api_pattern_attr_ignored,
-                              "import_owned", decl->getNameAsString());
+      ctx.Diags.diagnose({}, diag::api_pattern_attr_ignored, "import_owned",
+                         decl->getNameAsString());
     if (hasIteratorAPIAttr(cxxDecl))
-      desc.ctx.Diags.diagnose({}, diag::api_pattern_attr_ignored,
-                              "import_iterator", decl->getNameAsString());
+      ctx.Diags.diagnose({}, diag::api_pattern_attr_ignored, "import_iterator",
```

Comment:
This could not use `importerImpl.diagnose` with a proper `HeaderLoc`. I think this is a small enough change that could be included in this PR.